tests/test-dirs/expand_node/ppx-tests.t/run.t

```diff
@@ -0,0 +1,587 @@
+  [@@merlin.hide ]",
+      "deriver": {
+        "start": {
+          "line": 1,
+          "col": 15
+        },
+        "end": {
+          "line": 1,
+          "col": 34
+        }
+      }
+    },
+    "notifications": []
+  }
+
+Exception in structure
+  $ cat > apr.ml << EOF
+  > exception Foo of string [@@deriving rename]
+  > EOF
+
+  $ dune build
+
+  $ $MERLIN single expand-ppx -position 1:30 -filename ./apr.ml < ./apr.ml
+  {
+    "class": "return",
+    "value": {
+      "code": "include struct exception Foo_renamed of string  end[@@ocaml.doc \"@inline\"]
+  [@@merlin.hide ]",
+      "deriver": {
+        "start": {
+          "line": 1,
+          "col": 24
+        },
+        "end": {
+          "line": 1,
+          "col": 43
+        }
+      }
+    },
+    "notifications": []
+  }
+
+Exception in signature
+  $ cat > apr.mli << EOF
+  > exception Foo of string [@@deriving rename]
+  > EOF
+
+  $ dune build
+
+  $ $MERLIN single expand-ppx -position 1:30 -filename ./apr.mli < ./apr.mli
+  {
+    "class": "return",
+    "value": {
+      "code": "include sig [@@@ocaml.warning \"-32\"] exception Foo_renamed of string  end
+  [@@ocaml.doc \"@inline\"][@@merlin.hide ]",
+      "deriver": {
+        "start": {
+          "line": 1,
+          "col": 24
+        },
+        "end": {
+          "line": 1,
+          "col": 43
+        }
+      }
+    },
+    "notifications": []
+  }
+
+Module type declaration in structure
+  $ cat > apc.ml << EOF
+  > module type Stack = sig 
+  >   type t
+  >   type stack 
+  >   val empty : stack
+  >   val is_empty : stack -> bool
+  >   val push : t -> stack -> stack 
+  >   val pop : stack -> stack
+  >   val peek : stack -> t
+  > end [@@deriving rename]
+  > EOF
+
+  $ dune build
+
+  $ $MERLIN single expand-ppx -position 9:8 -filename ./apc.ml < ./apc.ml
+  {
+    "class": "return",
+    "value": {
+      "code": "include
+    struct
+      module type Stack_renamed  =
+        sig
+          type t
+          type stack
+          val empty : stack
+          val is_empty : stack -> bool
+          val push : t -> stack -> stack
+          val pop : stack -> stack
+          val peek : stack -> t
+        end
+    end[@@ocaml.doc \"@inline\"][@@merlin.hide ]",
+      "deriver": {
+        "start": {
+          "line": 9,
+          "col": 4
+        },
+        "end": {
+          "line": 9,
+          "col": 23
+        }
+      }
+    },
+    "notifications": []
+  }
+
+Module type declaration in signature
+  $ cat > apc.mli << EOF
+  > module type Stack = sig 
+  >   type t [@@deriving rename]
+  >   type stack 
+  >   val empty : stack
+  >   val is_empty : stack -> bool
+  >   val push : t -> stack -> stack 
+  >   val pop : stack -> stack
+  >   val peek : stack -> t
+  > end [@@deriving rename]
+  > EOF
+
+  $ dune build
+
+on attribute name deriving of type t
+TODO: a cursor here should only output the derived t and now the whole module
```

Comment:
What about this todo, is there something we can do ? Was this behavior already the one of the current LSP implementation ?

Reply:
The current LSP implementation will not give an output as it only works on `type_declarations`.